### PR TITLE
Distributed isomorphism

### PR DESCRIPTION
**Pull request by jessberg**
This makes a handlebars file that performs the isomorphism in a distributed manner.  Once it finds the target graph, it sends to storage, clears its isomorphism information, and stops propagating tracing information.

### CODE REVIEW

File: rewrite/main.rs [hunk 1]
```diff
@@ -133,11 +142,19 @@ fn main() {
...
                 codegen_simulator::CodeGenSimulator::generate_code_blocks(visitor_results, udfs);
-            write_to_handlebars(
-                &codegen_object,
-                bin_dir.join("filter.rs.handlebars"),
-                PathBuf::from(matches.value_of("output").unwrap()),
-            );
+            if distributed.is_none() {
+                write_to_handlebars(
+                    &codegen_object,
+                    bin_dir.join("filter.rs.handlebars"),
```
Comment (fruffy) [inline rewrite/main.rs:148]: Instead of having this large if-statement, just change the name, since that is apparently all that is necessary.

**Review by fruffy — CHANGES_REQUESTED** [review]

File: rewrite/main.rs (hunk 1 above)
Comment (jessberg) [reply rewrite/main.rs:148]: What do you mean?  Put it as an argument to write_to_handlebars and then do a similar if statement inside?

Comment (fruffy) [reply rewrite/main.rs:148]: "distributed_filter.rs.handlebars" vs "filter.rs.handlebars"

Comment (fruffy) [reply rewrite/main.rs:148]: The rest of the code looks the same to me.

Comment (jessberg) [reply rewrite/main.rs:148]: We want to support both at once though.  It should be an argument to main.

Comment (fruffy) [reply rewrite/main.rs:148]:
```rust
let handle_bar_str: &str;
if distributed.is_none() {
    handle_bar_str = "filter.rs.handlebars";
} else {
    handle_bar_str = "distributed_filter.rs.handlebars";
}
write_to_handlebars(
    &codegen_object,
    bin_dir.join(handle_bar_str),
    PathBuf::from(matches.value_of("output").unwrap()),
);
```

Comment (jessberg) [reply rewrite/main.rs]: Gotcha.  Should be fixed to that now.

**Review by fruffy — APPROVED** [review]

**Pull request merged**